framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -78,6 +84,21 @@ public abstract class CFAbstractStore<V extends CFAbstractValue<V>, S extends CF
      */
     protected Map<FieldAccess, V> fieldValues;
 
+    /**
+     * Fields in this set are never changed or removed by {@link
+     * #updateForMethodCall(MethodInvocationNode, CFAbstractValue)}.
+     *
+     * <p>E.g., this is useful for checkers that use an initialization type system and don't want
+     * monotonically initialized fields to become uninitialized by method calls.
+     *
+     * <p>Subcheckers should override {@link #isPersistent(FieldAccess)} instead of modifying this
+     * field directly.
+     *
+     * <p>We use this field as a cache instead of calling {@link #isPersistent(FieldAccess)} every
+     * time to avoid the performance issue in Issue1438.
+     */
+    protected final Set<FieldAccess> persistentFields;
```

Comment:
Should a new block be added to `internalVisualize` to also output this new information in the store?